lib/Dancer2/Core/Role/ConfigReader.pm

```diff
@@ -129,22 +129,23 @@ sub _build_config_files {
         }
     }
 
-    foreach my $ext (@exts) {
-        foreach my $file ( [ $location, "config.$ext" ],
-            [ $self->environments_location, "$running_env.$ext" ] )
-        {
-            my $path = path( @{$file} );
-            next if !-r $path;
-
-            push @files, $path;
+    # Keep order of files, even if they have different extensions
+    foreach my $file ( [ $location, "config" ],
+		       [ $self->environments_location, "$running_env" ] )
+      {
+	foreach my $ext (@exts) {
+	  my $path = path( $file->[0], $file->[1]. ".$ext" );
+	  next if !-r $path;
+
+	  push @files, $path;
         }
-    }
+      }
 
     # Look for *_local.ext files
     @files = map {
         (my $l = $_) =~ s/(\w+)(\.\w+)$/${1}_local$2/;
         $_, (-r $l ? $l : () );
-    } sort @files;
+    } @files;
```

Comment:
We should remove the `map` here by putting the logic for `_local.$ext` inside the nested `foreach` loops above.
(I'll clean this up in the next day or two!)